luminaire/optimization/hyperparameter_optimization.py

```diff
@@ -4,6 +4,7 @@
 import warnings
 warnings.filterwarnings('ignore')
 
+from ..utils import check_random_state
```

Comment:
All imports should be declared before `warnings`, you can move this to the first line.